enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+```sh
+$ opm bundle crds . # detect the mediatype of the current directory and write a YAML stream containing the provided CRDs to stdout
+$ opm bundle crds -o crds.yaml image://quay.io/my/bundle:v1.0.0 # pull and unpack the bundle image, then write the YAML stream to a file
+```
 
-OLM will support `registry+v2` bundles by synthesizing a CSV from its content. Resources usually generated by a CSV (e.g. APIService and Service) will not be directly applied, but instead be projected onto a synthesized CSV.
+Add a new command to `opm`, `opm bundle resources`, which takes as input an Operator Bundle and outputs a set of kubectl-able manifests that represent what OLM would apply to the cluster -- sans provided CRDs and dependencies-- for that bundle.
 
-This intermediary step (creating a CSV internally to represent the kubernetes objects) is a delibrate decision in 4.5 to help focus and deliver the feature. This may be relaxed in future release versions.
+_Ex._
 
-#### Validation
+```sh
+$ opm bundle resources . # detect the mediatype of the current directory, generate the resources to apply for it as OLM would, and write the YAML stream to stdout
+$ opm bundle resources -o my-bundle.yaml image://quay.io/my/bundle:v1.0.0 # pull and unpack the bundle image, generate the resources to apply for it as OLM would, and write the YAML stream to a file
+```
 
-The [operator-framework static validation package](https://github.com/operator-framework/api/tree/master/pkg/validation) can be extended and used to validate the content being fed into a `registry+v2` bundle build. New validators will be added to `opm` which ensure all required manifests and metadata are present and valid.
+#### Validate Bundles
 
-#### Indexes
+As a Pipeline Maintainer, I want a way to validate `registry+v2` bundles, so that I can ensure OLM is able to unpack and apply it before I add it to any release index.
 
-When a `registry+v2` bundle is added to an index, `opm` will also synthesize and add a CSV. A lot of the existing `opm` and `operator-registry` code assumes bundles always contain CSVs, which make synthesis the simplest path to supporting `registry+v2` bundles -- in the future, we may want to revisit this decision, especially when the [`Operator` API](https://github.com/openshift/enhancements/blob/master/enhancements/olm/simplify-apis.md#proposal) is released.
+##### UX
 
-To fill out the dependency graph between versions of bundles, OLM considers the "replaces" field on the CSV which points to the previous version of the operator. This field will need to be supplied in the additional metadata the operator author provides in the bundle so that OLM can build dependency graphs. There is another enhancement around semantic versioning that would remove this requirement altogether, but it is independent of this enhancement.
+Add a new command to `opm`, `opm bundle validate`, which takes as input an Operator Bundle and fails with informative errors should the bundle be invalid.
```

Comment:
Doesn't this already exist? why are we including it as part of the enhancement

Reply:
It's `opm alpha bundle validate` today.